binlog_streamer.go

```diff
@@ -146,7 +152,7 @@ func (s *BinlogStreamer) Run() {
 				"file": s.lastStreamedBinlogPosition.Name,
 			}).Info("rotated binlog file")
 		case *replication.RowsEvent:
-			err = s.handleRowsEvent(ev)
+			err = s.handleRowsEvent(ctx, ev)
```

Comment:
Do we really need to pass the context to the listener? I want reiterate that our cancellation policy is fairly relaxed

Reply:
One of the issues of not passing this down to the listeners is because we have listeners that have retries that could proceed to block for a long period of time after a cancel has been received.

This kind of blocking might become a concern during testing and is the ultimate reason why I put the cancellation all over the place. Perhaps we can limit our test designs such that those WithRetries calls to interact with the database do not fail.